### PR TITLE
[dga] Ensure event.kind is correctly set for pipeline errors

### PR DESCRIPTION
**Pull request by MakoWish**
## What does this PR do?

Modify dga integration to correctly set `event.kind` for pipeline errors and ensure `error.message` is an array.

## Checklist

- [X] I have reviewed tips for building integrations and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's changelog.yml file.
- [x] I have verified that Kibana version constraints are current according to guidelines.
 
## Author's Checklist

- [x]  

## Related issues

* For #6582

## Screenshots

### CODE REVIEW

File: packages/dga/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# newer versions go on top
+- version: "1.1.0"
+  changes:
```
Comment (sodhikirti07) [inline packages/dga/changelog.yml:3]: Shouldn't the version be `1.0.3`? Same for `manifest.yml`. 

cc: @ajosh0504

**Review by sodhikirti07 — APPROVED** [review]

**Comment by MakoWish** [comment]
FYI... As a contributor, I don't have permissions to merge.

**Comment by sodhikirti07** [comment]
@elastic/ml-ui Can someone approve this PR before we merge the changes?

**Review by alvarezmelissa87 — APPROVED** [review]
Code LGTM ⚡

**Pull request merged**